modules/calib3d/test/test_solvepnp_ransac.cpp

```diff
@@ -101,25 +222,41 @@ class CV_solvePnPRansac_Test : public cvtest::BaseTest
         const double maxVal = 1.0;
         rvec.create(3, 1, CV_64FC1);
         tvec.create(3, 1, CV_64FC1);
+
         for (int i = 0; i < 3; i++)
         {
             rvec.at<double>(i,0) = rng.uniform(minVal, maxVal);
-            tvec.at<double>(i,0) = rng.uniform(minVal, maxVal/10);
+            tvec.at<double>(i,0) = (i == 2) ? rng.uniform(minVal*10, maxVal) : rng.uniform(-maxVal, maxVal);
         }
     }
 
-    virtual bool runTest(RNG& rng, int mode, int method, const vector<Point3f>& points, const double* epsilon, double& maxError)
+    virtual bool runTest(RNG& rng, int mode, int method, const vector<Point3f>& points, double& errorTrans, double& errorRot)
     {
+        if ((!planar && method == SOLVEPNP_IPPE) || method == SOLVEPNP_IPPE_SQUARE)
+        {
+            return true;
+        }
+
         Mat rvec, tvec;
         vector<int> inliers;
         Mat trueRvec, trueTvec;
         Mat intrinsics, distCoeffs;
         generateCameraMatrix(intrinsics, rng);
-        if (method == 4) intrinsics.at<double>(1,1) = intrinsics.at<double>(0,0);
+        //UPnP is mapped to EPnP
```

Comment:
I have commented all the specific code for `UPnP`, `DLS?` since they are internally remapped to `EPnP`.
It is misleading to have different results, thresholds between `EPnP` and `UPnP`, `DLS`.